homeassistant/components/sensor/habitica.py

```diff
@@ -0,0 +1,151 @@
+"""
+The Habitica sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.habitica/
+"""
+
+import logging
+from collections import namedtuple
+# from typing import Any
+from datetime import timedelta
+import voluptuous as vol
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers import config_validation as cv
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+DOMAIN = "habitica"
+SENSOR_DATA_IDX = "sensor." + DOMAIN
+DEPENDENCIES = [DOMAIN]
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+
+CONF_NAME = "name"
+CONF_SENSORS_LIST = "sensors"
+ST = SensorType = namedtuple('SensorType', [
+    "name", "icon", "unit", "path"
+])
+
+SENSORS_TYPES = {
+    'name': ST('Name', None, '', ["profile", "name"]),
+    'hp': ST('HP', 'mdi:heart', 'HP', ["stats", "hp"]),
+    'maxHealth': ST('max HP', 'mdi:heart', 'HP', ["stats", "maxHealth"]),
+    'mp': ST('Mana', 'mdi:auto-fix', 'MP', ["stats", "mp"]),
+    'maxMP': ST('max Mana', 'mdi:auto-fix', 'MP', ["stats", "maxMP"]),
+    'exp': ST('EXP', 'mdi:star', 'EXP', ["stats", "exp"]),
+    'toNextLevel': ST(
+        'Next Lvl', 'mdi:star', 'EXP', ["stats", "toNextLevel"]),
+    'lvl': ST(
+        'Lvl', 'mdi:arrow-up-bold-circle-outline', 'Lvl', ["stats", "lvl"]),
+    'gp': ST('Gold', 'mdi:coin', 'Gold', ["stats", "gp"]),
+    'class': ST('Class', 'mdi:sword', '', ["stats", "class"])
+}
+ALL_SENSORS_TYPES = list(SENSORS_TYPES.keys())
+
+
+def ensure_list_of_option_types(value):
+    """Wrap value in list if it is not one."""
+    if value is None:
+        return ALL_SENSORS_TYPES
+    value = value if isinstance(value, list) else [value]
+    for item in value:
+        if item not in ALL_SENSORS_TYPES:
+            raise vol.Invalid(
+                "Found {}, expected one of {}".format(
+                    item, ALL_SENSORS_TYPES))
+    return value
+
+
+PLATFORM_SCHEMA = cv.PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_SENSORS_LIST): ensure_list_of_option_types
```

Comment:
Use `vol.All(cv.ensure_list, [vol.In(ALL_SENSORS_TYPES)])` instead, and put ALL_SENSORS_TYPES as default.